### PR TITLE
7.9.0 release notes

### PR DESCRIPTION
**Pull request by pugnascotia**
This PR adds release notes and release highlights for 7.9.0.

### Preview

https://elasticsearch_59961.docs-preview.app.elstc.co/diff

### CODE REVIEW

File: docs/java-rest/low-level/configuration.asciidoc [hunk 1]
```diff
@@ -25,7 +25,7 @@ seconds).
 include-tagged::{doc-tests}/RestClientDocumentation.java[rest-client-config-timeouts]
 --------------------------------------------------
 
-timeouts also can set by RequestOptions per request.this will override RestClient customizeRequestConfig.
+Timeouts also can set per request with RequestOptions. This will override RestClient customizeRequestConfig.
```
Comment (lcawl) [inline docs/java-rest/low-level/configuration.asciidoc:28]: Not sure what this is doing in with the release notes, but it looks like it's missing a word.
```suggestion
Timeouts also can be set per request with RequestOptions, which overrides RestClient customizeRequestConfig.
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 2]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+
+SQL::
+* Implement TIME_PARSE function for parsing strings into TIME values {es-pull}55223[#55223] (issues: {es-issue}54963[#54963], {es-issue}55095[#55095])
+* Implement TOP as an alternative to LIMIT {es-pull}57428[#57428] (issue: {es-issue}41195[#41195])
+* Implement TRIM function {es-pull}57518[#57518] (issue: {es-issue}41195[#41195])
+* Improve performances of LTRIM/RTRIM {es-pull}57603[#57603] (issue: {es-issue}57594[#57594])
+* Make CASTing string to DATETIME more lenient {es-pull}57451[#57451]
+* Redact credentials in connection exceptions {es-pull}58650[#58650] (issue: {es-issue}56474[#56474])
+* Relax parsing of date/time escaped literals {es-pull}58336[#58336] (issue: {es-issue}58262[#58262])
+* SQL/EQL: Add support for scalars within LIKE/RLIKE {es-pull}56495[#56495] (issue: {es-issue}55058[#55058])
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:211]:
```suggestion
* Add support for scalars within LIKE/RLIKE {es-pull}56495[#56495] (issue: {es-issue}55058[#55058])
```

File: docs/reference/release-notes/highlights.asciidoc [hunk 3]
```diff
@@ -38,3 +38,186 @@ Other versions:
...
+// tag::notable-highlights[]
+[float]
+=== Improve speed and memory usage of multi-bucket aggregations
+
+Before {minor-version}, many of our more complex aggregations made a simplifying
+assumption that required that they duplicate many data structures once per
+bucket that contained them. The most expensive of these weighed in at a
+couple of kilobytes each. So for an aggregation like:
+
+----
```
Comment (lcawl) [inline docs/reference/release-notes/highlights.asciidoc:109]: The CI checks are failing here as follows:

> 04:24:16 FAILURE: Build failed with an exception.
04:24:16 
04:24:16 * What went wrong:
04:24:16 Execution failed for task ':docs:buildRestTests'.
04:24:16 > reference/release-notes/highlights.asciidoc[109:123]: Snippet missing a language.

File: docs/reference/release-notes/highlights.asciidoc [hunk 4]
```diff
@@ -38,3 +38,186 @@ Other versions:
...
+// tag::notable-highlights[]
+[discrete]
+=== EQL
+
+EQL (Event Query Language) is a declarative language dedicated for identifying patterns and relationships between events.
+
+Consider using EQL if you:
+
+* Use {es} for threat hunting or other security use cases
+* Search time-series data or logs, such as network or system logs
```
Comment (lcawl) [inline docs/reference/release-notes/highlights.asciidoc:51]: Per our Style Guide, this term is never hyphenated
```suggestion
* Search time series data or logs, such as network or system logs
```

File: docs/reference/release-notes/highlights.asciidoc [hunk 5]
```diff
@@ -38,3 +38,186 @@ Other versions:
...
+
+An in-depth discussion of EQL in ES scope can be found at {es-issue}49581[#49581].
+// end::notable-highlights[]
+
+// tag::notable-highlights[]
+[discrete]
+=== Data streams
+
+A _data stream_ is a convenient, scalable way to ingest, search, and manage
+continuously generated time-series data. The provide a simpler way to split
```
Comment (lcawl) [inline docs/reference/release-notes/highlights.asciidoc:75]:
```suggestion
continuously generated time series data. The provide a simpler way to split
```

File: docs/reference/release-notes/highlights.asciidoc [hunk 6]
```diff
@@ -38,3 +38,186 @@ Other versions:
...
+* Use {es} for threat hunting or other security use cases
+* Search time-series data or logs, such as network or system logs
+* Want an easy way to explore relationships between events
+
+A good intro on EQL and its purpose is available
+https://www.elastic.co/blog/introducing-event-query-language[in this blog
+post]. See the <<eql,EQL in Elasticsearch>> documentaton for an in-depth
+explanation, and also the
+https://eql.readthedocs.io/en/latest/query-guide/index.html[language
+reference]
```
Comment (lcawl) [inline docs/reference/release-notes/highlights.asciidoc:59]:
```suggestion
reference].
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 7]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
   default, and will be removed in {es} 8.0. See the
   <<deprecate_auto_import_dangling_indices,migration notes>>.
   {es-pull}58176[#58176] {es-pull}58898[#58898] (issue: {es-issue}48366[#48366])
+
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:40]: I think a "coming::[7.9.0]" should be added at the top of this page too

File: docs/reference/release-notes/7.9.asciidoc [hunk 8]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+
+
+[[feature-7.9.0]]
+[float]
+=== New features
+
+Aggregations::
+* Add moving percentiles pipeline aggregation {es-pull}55441[#55441] (issue: {es-issue}49452[#49452])
+* Add Normalize Pipeline Aggregation {es-pull}56399[#56399] (issue: {es-issue}51005[#51005])
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:59]: not a big deal, but it seems strange that these are capitalized
```suggestion
* Add normalize pipeline aggregation {es-pull}56399[#56399] (issue: {es-issue}51005[#51005])
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 9]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+* Add Variable Width Histogram Aggregation {es-pull}42035[#42035] (issues: {es-issue}9572[#9572], {es-issue}50863[#50863])
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:60]: Likewise, capitalization is not required:
```suggestion
* Add variable width histogram aggregation {es-pull}42035[#42035] (issues: {es-issue}9572[#9572], {es-issue}50863[#50863])
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 10]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+* Pipeline Inference Aggregation {es-pull}58193[#58193]
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:61]:
```suggestion
* Add pipeline inference aggregation {es-pull}58193[#58193]
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 11]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+* Speed up time interval arounding around dst {es-pull}56371[#56371] (issue: {es-issue}55559[#55559])
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:62]:
```suggestion
* Speed up time interval arounding around daylight savings time (DST) {es-pull}56371[#56371] (issue: {es-issue}55559[#55559])
```

File: docs/reference/release-notes/highlights.asciidoc [hunk 12]
```diff
@@ -38,3 +38,186 @@ Other versions:
...
+// end::notable-highlights[]
+
+
+// tag::notable-highlights[]
+[float]
+=== Wildcard field optimised for wildcard queries
+
+{es} now supports a `wildcard` field type, which stores values optimised
+for wildcard grep-like queries. While such queries are possible with other
+field types, they suffer from constraint that limit their usefulness.
```
Comment (debadair) [inline docs/reference/release-notes/highlights.asciidoc:197]:
```suggestion
field types, they suffer from constraints that limit their usefulness.
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 13]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+
+Geo::
+* Override doc_value parameter in Spatial XPack module {es-pull}53286[#53286] (issue: {es-issue}37206[#37206])
+
+Machine Learning::
+* Introduce DataFrameAnalyticsConfig update API {es-pull}58302[#58302] (issue: {es-issue}45720[#45720])
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:68]:
```suggestion
* Add update data frame analytics jobs API {es-pull}58302[#58302] (issue: {es-issue}45720[#45720])
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 14]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+=== Enhancements
+
+Aggregations::
+* Add support for numeric range keys {es-pull}56452[#56452] (issue: {es-issue}56402[#56402])
+* Added standard deviation / variance sampling to extended stats {es-pull}49782[#49782] (issue: {es-issue}49554[#49554])
+* Give significance lookups their own home {es-pull}57903[#57903]
+* Increase search.max_buckets to 65,535 {es-pull}57042[#57042] (issue: {es-issue}51731[#51731])
+* Optimize date_histograms across daylight savings time {es-pull}55559[#55559]
+* Return clear error message if aggregation type is invalid {es-pull}58255[#58255] (issue: {es-issue}58146[#58146])
+* Save memory on numeric sig terms when not top {es-pull}56789[#56789] (issue: {es-issue}55873[#55873])
```
Comment (debadair) [inline docs/reference/release-notes/7.9.asciidoc:91]:
```suggestion
* Save memory on numeric significant terms when not top {es-pull}56789[#56789] (issue: {es-issue}55873[#55873])
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 15]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+* Add Variable Width Histogram Aggregation {es-pull}42035[#42035] (issues: {es-issue}9572[#9572], {es-issue}50863[#50863])
+* Pipeline Inference Aggregation {es-pull}58193[#58193]
+* Speed up time interval arounding around dst {es-pull}56371[#56371] (issue: {es-issue}55559[#55559])
+
+Geo::
+* Override doc_value parameter in Spatial XPack module {es-pull}53286[#53286] (issue: {es-issue}37206[#37206])
+
+Machine Learning::
+* Introduce DataFrameAnalyticsConfig update API {es-pull}58302[#58302] (issue: {es-issue}45720[#45720])
+* Introduce ModelPlotConfig.annotations_enabled setting {es-pull}57539[#57539] (issue: {es-issue}55781[#55781])
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:69]:
```suggestion
* Introduce model_plot_config.annotations_enabled setting for anomaly detection jobs {es-pull}57539[#57539] (issue: {es-issue}55781[#55781])
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 16]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+
+CRUD::
+* Retry failed replication due to transient errors {es-pull}55633[#55633]
+
+Engine::
+* Don't log on RetentionLeaseSync error handler after an index has been deleted {es-pull}58098[#58098] (issue: {es-issue}57864[#57864])
+
+Features/Data streams::
+* Add support for snapshot and restore to data streams {es-pull}57675[#57675] (issues: {es-issue}53100[#53100], {es-issue}57127[#57127])
+* DataStream creation validation allows for prefixed indices {es-pull}57750[#57750] (issue: {es-issue}53100[#53100])
```
Comment (debadair) [inline docs/reference/release-notes/7.9.asciidoc:132]:
```suggestion
* Data stream creation validation allows for prefixed indices {es-pull}57750[#57750] (issue: {es-issue}53100[#53100])
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 17]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+* Support multiple tokens on LHS in stemmer_override rules (#56113) {es-pull}56484[#56484] (issue: {es-issue}56113[#56113])
+
+Authentication::
+* Add http proxy support for OIDC realm {es-pull}57039[#57039] (issue: {es-issue}53379[#53379])
+* Improve threadpool usage and error handling for API key validation {es-pull}58090[#58090] (issue: {es-issue}58088[#58088])
+* Support handling LogoutResponse from SAML idP {es-pull}56316[#56316] (issues: {es-issue}40901[#40901], {es-issue}43264[#43264])
+
+Authorization::
+* Add cache for application privileges {es-pull}55836[#55836] (issue: {es-issue}54317[#54317])
+* Add monitor and view_index_metadata to built-in `kibana_system` role {es-pull}57755[#57755]
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:118]:
```suggestion
* Add monitor and view_index_metadata privileges to built-in `kibana_system` role {es-pull}57755[#57755]
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 18]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+* Add RequestConfig support to RequestOptions {es-pull}57972[#57972]
+
+Infra/Circuit Breakers::
+* Enhance real memory circuit breaker with G1 GC {es-pull}58674[#58674] (issue: {es-issue}57202[#57202])
+
+Infra/Core::
+* Introduce node.roles setting {es-pull}54998[#54998]
+
+Infra/Packaging::
+* Remove DEBUG-level logging from actions in docker {es-pull}57389[#57389] (issues: {es-issue}51198[#51198], {es-issue}51459[#51459])
```
Comment (debadair) [inline docs/reference/release-notes/7.9.asciidoc:168]:
```suggestion
* Remove DEBUG-level logging from actions in Docker {es-pull}57389[#57389] (issues: {es-issue}51198[#51198], {es-issue}51459[#51459])
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 19]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+* Add data stream support to searchable snapshot action {es-pull}57873[#57873] (issue: {es-issue}53100[#53100])
+* Add data stream support to the shrink action {es-pull}57616[#57616] (issue: {es-issue}53100[#53100])
+* Add support for rolling over data streams  {es-pull}57295[#57295] (issues: {es-issue}53100[#53100], {es-issue}53488[#53488])
+* Check the managed index is not a data stream write index {es-pull}58239[#58239] (issue: {es-issue}53100[#53100])
+
+Features/Indices APIs::
+* Add default composable templates for new indexing strategy {es-pull}57629[#57629] (issue: {es-issue}56709[#56709])
+* Add index block api {es-pull}58094[#58094]
+* Add new flag to check whether alias exists on remove {es-pull}58100[#58100]
+* Add prefer_v2_templates parameter to Reindex {es-pull}56253[#56253] (issue: {es-issue}53101[#53101])
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:146]:
```suggestion
* Add prefer_v2_templates parameter to reindex {es-pull}56253[#56253] (issue: {es-issue}53101[#53101])
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 20]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+* Add description to submit and get async search, as well as cancel tasks {es-pull}57745[#57745]
+* Add matchBoolPrefix static method in query builders {es-pull}58637[#58637] (issue: {es-issue}58388[#58388])
+* Add range query support to wildcard field {es-pull}57881[#57881] (issue: {es-issue}57816[#57816])
+* Group docIds by segment in FetchPhase to better use LRU cache {es-pull}57273[#57273]
+* Improve error handling when decoding async execution ids {es-pull}56285[#56285]
+* Specify reason whenever async search gets cancelled {es-pull}57761[#57761]
+* Use index sort range query when possible. {es-pull}56657[#56657] (issue: {es-issue}48665[#48665])
+
+Security::
+* Add ML admin permissions to the kibana_system role {es-pull}58061[#58061]
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:223]:
```suggestion
* Add machine learning admin permissions to the kibana_system role {es-pull}58061[#58061]
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 21]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+* Just log 401 stacktraces {es-pull}55774[#55774]
+
+Snapshot/Restore::
+* Deduplicate Index Metadata in BlobStore {es-pull}50278[#50278] (issues: {es-issue}45736[#45736], {es-issue}46250[#46250], {es-issue}49800[#49800])
+* Default to zero replicas for searchable snapshots {es-pull}57802[#57802] (issue: {es-issue}50999[#50999])
+* Enable Fully Concurrent Snapshot Operations {es-pull}56911[#56911]
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:229]:
```suggestion
* Enable fully concurrent snapshot operations {es-pull}56911[#56911]
```

**Review by debadair — APPROVED** [review]
Left a few minor suggestions, but otherwise LGTM.

File: docs/reference/release-notes/7.9.asciidoc [hunk 22]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+* Reword INDEX_READ_ONLY_ALLOW_DELETE_BLOCK message {es-pull}58410[#58410] (issues: {es-issue}42559[#42559], {es-issue}50166[#50166], {es-issue}58376[#58376])
+
+Authentication::
+* Map only specific type of OIDC Claims {es-pull}58524[#58524]
+
+Authorization::
+* Change privilege of enrich stats API to monitor {es-pull}52027[#52027] (issue: {es-issue}51677[#51677])
+
+Engine::
+* Fix Local Translog Recovery not Updating Safe Commit in Edge Case {es-pull}57350[#57350] (issue: {es-issue}57010[#57010])
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:267]:
```suggestion
* Fix local translog recovery not updating safe commit in edge case {es-pull}57350[#57350] (issue: {es-issue}57010[#57010])
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 23]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+
+Features/Ingest::
+* Fix ingest simulate verbose on failure with conditional {es-pull}56478[#56478] (issue: {es-issue}56004[#56004])
+
+Geo::
+* Check for degenerated lines when calculating the centroid {es-pull}58027[#58027] (issue: {es-issue}55851[#55851])
+* Fix bug in circuit-breaker check for geoshape grid aggregations {es-pull}57962[#57962] (issue: {es-issue}57847[#57847])
+
+Infra/Scripting::
+* Fix Source Return Bug in Scripting {es-pull}56831[#56831] (issue: {es-issue}52103[#52103])
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:285]:
```suggestion
* Fix source return bug in scripting {es-pull}56831[#56831] (issue: {es-issue}52103[#52103])
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 24]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+SQL::
+* Introduce JDBC option for meta pattern escaping {es-pull}40661[#40661] (issue: {es-issue}40640[#40640])
+
+Search::
+* Don't omit empty arrays when filtering _source {es-pull}56527[#56527] (issues: {es-issue}20736[#20736], {es-issue}22593[#22593], {es-issue}23796[#23796])
+* Fix casting of scaled_float in sorts {es-pull}57207[#57207]
+
+Snapshot/Restore::
+* Account for recovery throttling when restoring snapshot {es-pull}58658[#58658] (issue: {es-issue}57023[#57023])
+* Fix Noisy Logging during Snapshot Delete {es-pull}56264[#56264]
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:304]:
```suggestion
* Fix noisy logging during snapshot delete {es-pull}56264[#56264]
```

File: docs/reference/release-notes/7.9.asciidoc [hunk 25]
```diff
@@ -37,3 +37,281 @@ Dangling indices::
...
+* Fix S3ClientSettings Leak {es-pull}56703[#56703] (issue: {es-issue}56702[#56702])
```
Comment (lcawl) [inline docs/reference/release-notes/7.9.asciidoc:305]:
```suggestion
* Fix S3ClientSettings leak {es-pull}56703[#56703] (issue: {es-issue}56702[#56702])
```

File: docs/java-rest/low-level/configuration.asciidoc (hunk 1 above)
Comment (pugnascotia) [reply docs/java-rest/low-level/configuration.asciidoc:28]: Ah, I meant to mention that - in tracking down what one of the PRs was for, I found a change to the documentation that needed some improvement (but didn't improve enough 😁).

File: docs/reference/release-notes/7.9.asciidoc (hunk 7 above)
Comment (pugnascotia) [reply docs/reference/release-notes/7.9.asciidoc:42]: Added this.

File: docs/reference/release-notes/highlights.asciidoc (hunk 3 above)
Comment (pugnascotia) [reply docs/reference/release-notes/highlights.asciidoc:110]: I added "[source,console]", hopefully that fixes CI.

**Review by lcawl — APPROVED** [review]
LGTM, thanks!

**Pull request merged**